### ISSUE TITLE
Baudrate for serial2 is ignored

### ISSUE THREAD

**Issue by wiiverueckt** [MegaPirateNG/ardupilot-mpng#34]
the boud rate on seral2 GPS is still 38400 even if I change it to 115200.
I solved the probem by changing the gps speed to 38400.

**Comment by SirAlex** [comment]
This already fixed in latest R3

**Comment by wiiverueckt** [comment]
I have that issue with ArduCopter V3.1-rc4 MPNG-R1b.

**Comment by smurfy** [comment]
Sadly i think no.

Background:
Appart of owning a Crius V2 for my Plane i also own a Witespy Ez3 with GPS.
Witespy's GPS uses 115200 and 5hz instead of 38400 and 1hz.
I did some tests yesterday with UBLOX but did not get it working.

While trying to figure out whats going on i noticed that in the `AP_GPS_UBLOX.cpp` the baudrate seams hardcoded.

```
AP_GPS_UBLOX::_configure_gps(void)
{
    struct ubx_cfg_nav_rate msg;
    const unsigned baudrates[4] = {9600U, 19200U, 38400U, 57600U};

    // the GPS may be setup for a different baud rate. This ensures
    // it gets configured correctly
    for (uint8_t i=0; i<4; i++) {
        _port->begin(baudrates[i]);
        _write_progstr_block(_port, _ublox_set_binary, _ublox_set_binary_size);
        while (_port->tx_pending()) {
          hal.scheduler->delay(1);
        }
    }
    _port->begin(38400U);

    // ask for navigation solutions every 200ms
    msg.measure_rate_ms = 200;
    msg.nav_rate        = 1;
    msg.timeref         = 0;     // UTC time
    _send_message(CLASS_CFG, MSG_CFG_RATE, &msg, sizeof(msg));

    // ask for the messages we parse to be sent on every navigation solution
    _configure_message_rate(CLASS_NAV, MSG_POSLLH, 1);
    _configure_message_rate(CLASS_NAV, MSG_STATUS, 1);
    _configure_message_rate(CLASS_NAV, MSG_SOL, 1);
    _configure_message_rate(CLASS_NAV, MSG_VELNED, 1);

    // ask for the current navigation settings
    Debug("Asking for engine setting\n");
    _send_message(CLASS_CFG, MSG_CFG_NAV_SETTINGS, NULL, 0);
}
```

**Comment by SirAlex** [comment]
Problem solved in **3.0.1-R3** and not in 3.1-rc4 (I'm fixed it after last commit into master, so 3.1 doesn't have this fix).
But changing port speed not affected on AUTO gps driver. If you change speed, then you must change driver to NMEA, UBLOX, MTK... etc, but not AUTO.
